src/components/settings/connection-settings-card.tsx

```diff
@@ -70,6 +70,24 @@ const UpstreamProxyDropdown = styled(Select)`
     margin-right: 10px;
 `;
 
+const HOST_PATTERN = '[A-Za-z0-9\\-.]+';
+const PORT_PATTERN = '\\d+';
+// Disallow "@", "/" and ":" in username/password
+const CRED_PATTERN = '[^@/:]+';
+
+const PROXY_HOST_REGEXES = [
+    // 0) host or host:port
+    new RegExp(`^(${HOST_PATTERN})(:${PORT_PATTERN})?$`),
+    // 1) user:pass@host:port
+    new RegExp(`^(${CRED_PATTERN}):(${CRED_PATTERN})@(${HOST_PATTERN}):(${PORT_PATTERN})$`),
+    // 2) host:port@user:pass
+    new RegExp(`^(${HOST_PATTERN}):(${PORT_PATTERN})@(${CRED_PATTERN}):(${CRED_PATTERN})$`),
+    // 3) host:port:user:pass
+    new RegExp(`^(${HOST_PATTERN}):(${PORT_PATTERN}):(${CRED_PATTERN}):(${CRED_PATTERN})$`),
+    // 4) user:pass:host:port
+    new RegExp(`^(${CRED_PATTERN}):(${CRED_PATTERN}):(${HOST_PATTERN}):(${PORT_PATTERN})$`),
```

Comment:
Apparently `user@host:port` also exists (implicitly empty password) and isn't that rare. I'm pretty sure that works now so it would be good not to break it.

Technically `user:@...`, `:password@...` and even `:@...` are valid as well but I think we can probably ignore those as too weird to appear in practice, and just wait & see if anybody complains.